.travis.yml

```diff
@@ -9,107 +10,59 @@ install:
-
-    - stage: Run tests on oldest set of dependencies (Python 3.5.3)
-      env: NUMPY="1.12.0", PANDAS="0.20.3", SCIKITLEARN="0.19.0", DASK="0.15.2", DISTRIBUTED="1.18.3", SCIPY="1.2.0"
-      before_script:
-        - pip install numpy==1.12.0  # First version with official python 3.6 support.
-        - pip install pandas==0.20.3
-        - pip install scikit-learn==0.19.0
-        - pip install dask==0.15.2
-        - pip install distributed==1.18.3
-        # need to downgrade tornado manually
-        - pip install tornado==4.5.3
-        - pip install scipy==1.2.0
-        - pip list
-      script:
-        - sed -i 's/\-n auto/\-n 2/g' setup.cfg
-        - "if [ $TRAVIS_PULL_REQUEST == false ] && ! [ $TRAVIS_BRANCH == 'master' ]; then pytest tests/units -n2; else pytest tests; fi"
+      script: *test
+    - env: NUMPY="1.12.0", PANDAS="0.20.3", SCIKITLEARN="0.19.0", DASK="0.15.2", DISTRIBUTED="1.18.3", SCIPY="1.2.0"
+      python: 3.6
+      script: *test
+    - env: NUMPY="1.12.0", PANDAS="0.20.3", SCIKITLEARN="0.19.0", DASK="0.15.2", DISTRIBUTED="1.18.3", SCIPY="1.2.0"
       python: 3.5.3
+      script: *test
 
-    - stage: Run tests on oldest set of dependencies (Python 3.6)
-      env: NUMPY="1.12.0", PANDAS="0.20.3", SCIKITLEARN="0.19.0", DASK="0.15.2", DISTRIBUTED="1.18.3", SCIPY="1.2.0"
-      before_script:
-        - pip install numpy==1.12.0  # First version with official python 3.6 support.
-        - pip install pandas==0.20.3
-        - pip install scikit-learn==0.19.0
-        - pip install dask==0.15.2
-        - pip install distributed==1.18.3
-        # need to downgrade tornado manually
-        - pip install tornado==4.5.3
-        - pip install scipy==1.2.0
-        - pip list
-      script:
-        - sed -i 's/\-n auto/\-n 2/g' setup.cfg
-        - "if [ $TRAVIS_PULL_REQUEST == false ] && ! [ $TRAVIS_BRANCH == 'master' ]; then pytest tests/units -n2; else pytest tests; fi"
-      python: 3.6
+    - stage: Test coverage
+      if: env(PYTEST_ADDOPTS) = "-c setup.cfg" AND fork = false
+      script: coveralls
 
-    - stage: Run tests on oldest set of dependencies (Python 3.7)
-      env: NUMPY="1.15.1", PANDAS="0.23.2", SCIKITLEARN="0.19.2", DASK="0.16.1", DISTRIBUTED="1.18.3", SCIPY="1.2.0"
-      before_script:
-        - pip install numpy==1.15.1 # First version with official python 3.7 support.
-        - pip install pandas==0.23.2 # First version with official python 3.7 support.
-        - pip install scikit-learn==0.19.2 # First version with python 3.7 support.
-        - pip install dask==0.16.1
-        - pip install distributed==1.18.3
-        # need to downgrade tornado manually
-        - pip install tornado==4.5.3
-        - pip install scipy==1.2.0
-        - pip list
-      script:
-        - sed -i 's/\-n auto/\-n 2/g' setup.cfg
-        - "if [ $TRAVIS_PULL_REQUEST == false ] && ! [ $TRAVIS_BRANCH == 'master' ]; then pytest tests/units -n2; else pytest tests; fi"
+    - stage: Deploy
```

Comment:
Again there is probably a good reason, by why not just have this as the deploy option (not as a stage)?

Reply:
Okay, I am not a travis expert, so please take explanations below with a grain of salt.

If you look at current travis, there are 6 different stages.  As soon as a stage fail, others are cancelled, as can be seen at https://travis-ci.org/blue-yonder/tsfresh/builds/616016632 or https://travis-ci.org/blue-yonder/tsfresh/builds/614728433

When something goes wrong, it is important to gather as much information about all builds as possible, so this behavior is IMO very annoying.

I tried to understand why you did it that way, and thought that you wanted to avoid job concurrency.  Also I do not fully understand how coverage works; for instance if you look at https://travis-ci.org/blue-yonder/tsfresh/builds/616693314/ config says that coveralls is called by the build with python 3.7 and latest dependencies.  But if you look at all logs, you see at the end no-data-collected warning in all builds but 2343.4, which is for python 3.5.3 with oldest dependencies.  I was really puzzled, and tried something different which seems to work as expected ;)
So unless you have an explanation for current logs, I prefer this solution with stages.

Reply:
I forgot to mention that by moving deployment to a different stage, this ensures that all builds succeeded, and not only python 3.7 with latest dependencies.

Reply:
:+1: